enhancements/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,239 @@
+   marked as `provisional` as an idea worth exploring in the future, but not
+   accepted as ready to execute. Anything marked as `implementable` should
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall feature summary is present in the [Management workload partitioning](management-workload-partitioning.md) feature. This document focuses on how the feature enablement API and its interactions will look like.
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### Goals
+
+This feature is about designing a read-only API that will describe the enabled workload partitions (types, classes, etc.). This information is needed for kubelet to start exposing the right resources as well for the admission webhook to know when the pod manipulation is needed.
+
+It is expected this API will be created at the installation process. Either manually or using the Performance addon operator render mode.
+
+The short term goal is to get enough functionality to create the `management` partition where all OpenShift infrastructure could be placed. This partition is cluster wide and only affects pods with certain pre-agreed-upon labels and placed in specific pre-annotated namespaces.
+
+### Non-Goals
+
+- Long term goal of allowing cluster admins to create their own partitions is not part of this proposal for now
+- Day 2 creation of partitions is not part of the proposal
+- Namespace or MCP specific partitions are not part of the proposal
+
+## Proposal
+
+The proposal is to define a new cluster-wide Custom Resource Definition that would describe the allowed partition names in the status section. That way it hints at being a read only object where no user/admin input or modifications are expected.
+
+```yaml
+apiVersion: workloadpartitioning.openshift.io/v1
```

Comment:
In the original enhancement we use `workload.openshift.io` in some annotations. Should we do that here, too, and make this something like `partitioning.workload.openshift.io` What is standard/preferred for OpenShift APIs?

Alternatively, do we anticipate other API inputs related to workloads that might live on this CR later, so we should not include "partition" in the name?